File: extensions/youtube/src/main/java/app/morphe/extension/youtube/patches/RemoveViewerDiscretionDialogPatch.java
package app.morphe.extension.youtube.patches;

import app.morphe.extension.youtube.settings.Settings;

@SuppressWarnings("unused")
public class RemoveViewerDiscretionDialogPatch {

    /**
     * Injection point.
     */
    public static boolean hideViewDiscretionDialog(boolean originalValue) {
        if (Settings.REMOVE_VIEWER_DISCRETION_DIALOG.get()) {
            return true;
        }
        return originalValue;
    }
}

File: patches/src/main/kotlin/app/morphe/patches/youtube/interaction/dialog/RemoveViewerDiscretionDialogPatch.kt
package app.morphe.patches.youtube.interaction.dialog

import app.morphe.patches.shared.interaction.dialog.removeViewerDiscretionDialogPatch
import app.morphe.patches.youtube.misc.extension.sharedExtensionPatch
import app.morphe.patches.youtube.misc.settings.PreferenceScreen
import app.morphe.patches.youtube.misc.settings.settingsPatch
import app.morphe.patches.youtube.shared.Constants.COMPATIBILITY_YOUTUBE

@Suppress("unused")
val removeViewerDiscretionDialogPatch = removeViewerDiscretionDialogPatch(
    block = {
        dependsOn(
            sharedExtensionPatch,
            settingsPatch,
        )

        compatibleWith(COMPATIBILITY_YOUTUBE)
    },
    preferenceScreen = PreferenceScreen.GENERAL
)
